Twrp 8.1

csather79 opened the pull request:
Merge please

nkk71 committed:
remove-minimal.xml: Initial commit (removes everything!)

nkk71 committed:
remove-minimal.xml: Don't remove needed repos

nkk71 committed:
Exclude bootable/recovery

nkk71 committed:
remove-minimal.xml: Partial changes needed for FBE decryption

* This is only a partial commit. Some repos still need to be
  included manually, see the comment about android_frameworks_base.
  And some additional libs/dirs are needed from AOSP r34 source
  since they are not yet available (or updated) in the current omni
  repo at the time of this writing.

* Using AOSP android-8.0.0_r34 tag:

  For at least a build:
    Add: frameworks/base/core/java/android/content
    Add: frameworks/base/core/java/android/security

  For weaver:
    Add: hardware/interfaces/weaver

  For libkeymaster1.so -> libkeymaster_portable.so and libkeymaster_staging.so
    Replace: system/keymaster
    Replace: system/security/keystore

  For libkeyutils.so
    Add: system/core/libkeyutils

nkk71 committed:
Minimal TWRP: Add/Replace needed libs for FBE as well as busybox

* Addresses the problems mentioned in
  https://github.com/nkk71/android/commit/70da2068a8b63758c776b4eed663feaf5a55f6cd
  1) system/keymaster and system/security are replaced by
     AOSP-android-8.0.0_r34 versions
  2) weaver and libkeyutils get synced to a staging directory
     then symlinked to the proper destination directories

* Comment out the removal of android_frameworks_base

* Add TeamWin's android-8.0 busybox

nkk71 committed:
Add TWRP repo from omnirom

nkk71 committed:
Remove repos I clone myself

nkk71 committed:
Change 'staging' directory for TWRP extras

* .repo and .git are excluded from an "include
  all subdir makefiles" so use the
  Teamwin/.repo/... to avoid module/lib/etc
  'collisions'.

nkk71 committed:
Add platform/frameworks/support

* platform/frameworks/support is needed by
  android_frameworks_base.

* When building with a stripped down framework/base
  it was not needed. For the 'stripped' information
  refer to my comments here https://github.com/nkk71/android/commit/70da2068a8b63758c776b4eed663feaf5a55f6cd

nkk71 committed:
Revert: Add platform/frameworks/support

* platform/frameworks/support is needed by
  android_frameworks_base.

* When building with a stripped down framework/base
  it was not needed. For the 'stripped' information
  refer to my comments here https://github.com/nkk71/android/commit/70da2068a8b63758c776b4eed663feaf5a55f6cd

(reverted from commit 699490ddcfb74f8f62e8c5d5e9063951ff5378b9)

nkk71 committed:
Merge remote-tracking branch 'omnirom/android-8.0' into android-8.0-minimal

nkk71 committed:
remove-minimal: Update for upstream omni changes

nkk71 committed:
Update readme

nkk71 committed:
Merge remote-tracking branch 'omnirom/android-8.1' into test-8.1-mine

# Conflicts:
#	README.md
#	default.xml
#	omni-default.xml

nkk71 committed:
Minimal TWRP manifest: Updates for omni-8.1

* PLEASE READ THIS:
  Due to way the build environment now checks every
  single makefile, whether used or unused you will
  have to set the following variable:
      export ALLOW_MISSING_DEPENDENCIES=true
  as mentioned by the build message:
  "You can set ALLOW_MISSING_DEPENDENCIES=true in your environment
  if this is intentional, but that may defer real problems until
  later in the build."

nkk71 committed:
Update Readme

maxwen committed:
manifest: switch remaning apps to 8.1

Change-Id: I5b7a78be7e73c8e0d43d2eab415ba8f37f3cd8ad

maxwen committed:
manifest: omni CellBroadcastReceiver and EmergencyInfo

thanks google we need to fork this just to fix your lazyness

Change-Id: I5626826c607edd818c531d16ae4974dc53ad081e

darkobas committed:
manifest: add omni Telephony repos

Change-Id: I1536e253b736b136bcb74fb662c8fa0072f10400

maxwen committed:
manifest: add omni Contacts repo

Change-Id: I621c2df8a2751f240411d96a869c1622f62e83b9

maxwen committed:
manifest: add omni Messaging repo

Change-Id: I38e93a1031e3cc0f269e23becbbe38cd8ec393cc

maxwen committed:
android: add back ntfs-3g repo

Change-Id: Ic0ddba439b22e789e441f0507521a5e5dde3fedc

darkobas committed:
manifest: change android revision to android-8.1.0_r7

Change-Id: I14961838c20e33c1bf61821754a2437da10e18dd

darkobas committed:
manifest: add system/extras

Change-Id: I73f242e1735c9e28dc990bacaefe554ab915a882

CaptainThrowback committed:
Update minimal manifest for latest Omni changes

Change-Id: Ic3030a30bdf940a0826999833e0eae879d3d8d00

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

Conflicts:
	default.xml
	omni-default.xml

Change-Id: I3113e86ff38db6129c120cd1acf843570e530d76

CaptainThrowback committed:
Use Omni busybox

Has been updated for 8.1 tree

Change-Id: I01f89626f7bd2a02fd1ee63b6fae72c4225d8a9c

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

Change-Id: I68b4dea4d1a43d12a313b8d4300b1a6721253d5e

CaptainThrowback committed:
Update minimal manifest

Change-Id: I25bad56bad216c9fbc5a91978cc2a5eeac30f403

CaptainThrowback committed:
Revert "Use Omni busybox"

This reverts commit ba1ad7ae95ec4d3f87a9dfb0e6e903f061ae0b9d.

Still causes some segfaults

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

Change-Id: I83d82856a8baa8fb6e8d9a2c5e8397cccf02a8dd

Chippa-a committed:
manifest: Remove duplicate

Already in remove.xml.

fatal: remove-project element specifies non-existent project:
platform/packages/apps/Stk

Change-Id: Iec36eb51fb44fff8f13bd070694e41a8c4038006

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

Change-Id: Iea06e7c92f5c344c7021cbe3d681ee2e6a42f5ad

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

Change-Id: I41e32231716d62432e9167a28a6e1e43a16f5a29

imasaru committed:
Fixed clerical errors in README.md

Commit:
manifest: aosp: Fix mismerge

Change-Id: I4b0df41082f74ba2153046b58b7d1e5565d1381a
Signed-off-by: Humberto Borba <humberos@omnirom.org>

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

bigbiff committed:
twrp-8.1: DTC is needed for compile dtbs.

CaptainThrowback committed:
Revert "Revert "Use Omni busybox""

This reverts commit 0107c309b788761180d254f2a6d3e4bd0fc31466.

CaptainThrowback committed:
Update default TWRP branch to android-9.0

CaptainThrowback committed:
Merge remote-tracking branch 'omnirom/android-8.1' into twrp-8.1

CaptainThrowback committed:
remove-minimal: libxml2 needed for some devices

CaptainThrowback committed:
Add external_magisk-prebuilt repo to TWRP manifest

CaptainThrowback committed:
manifest: switch to TeamWin vendor_omni repo

- Changes for TWRP will be pushed here rather than OmniROM
- Add in brotli and bsdiff repos so update-binary can be built
- Add TeamWin remote for better handling
- Cleanup

Change-Id: I63ef2686930b90205be3b36c71039cccba4bbc7d

CaptainThrowback committed:
manifest: allow update_engine_sideload to be built

Change-Id: I9f130b398b40695b624b6b8da1dd696a6854e956

MartinX3 commented:
Did you ever read the [README.MD](https://github.com/omnirom/android/blob/android-8.1/README.md) file?

And what's your intention?

CaptainThrowback committed:
twrp: source repo from TeamWin

Change-Id: I196949e6822f38b66167745c262b915c0139770d

CaptainThrowback committed:
twrp-extras: use TeamWin update_engine repo

Change-Id: I2efa3b2ef6399e146ec9f35b63178fe13a4a8da2

CaptainThrowback committed:
twrp-extras: use TeamWin boringssl repo

Change-Id: Ia842f31e0dc7628f17052b88d66a15087602dc00

CaptainThrowback committed:
twrp-extras: fix update_engine repo source

- Add TeamWin build repo (commented out for now)

Change-Id: Ibd8556d04e71a9fc4460e8d88c21ae7db4e61491

CaptainThrowback committed:
Update branch on magisk-prebuilt repo

CaptainThrowback committed:
twrp-8.1: add repos for nano

Change-Id: Ic639d8c8096af5f4251de1d3f158f1994316ec57

CaptainThrowback committed:
twrp-8.1: add repo for bash

Change-Id: Ice6316202994c70cf91534152d2451e178c14af3

CaptainThrowback committed:
twrp-8.1: sync a shallow clone of frameworks folders

".repo" is no longer allowed in folder name in newer versions of repo,
so we now need to sync some frameworks folders and dependencies
for builds to run/finish successfully.

This adds ~1GB to the overall downloaded source, but there is no
workaround for this anymore since repo was updated.

Change-Id: Ica996ad5e9e8595868011c4246d479eac5777b48

MartinX3 committed:
Update README.md

CaptainThrowback committed:
Use TeamWin android_build repo

Change-Id: I5420465a259e6c007199082272757cf7b8ca957f

CaptainThrowback committed:
Add python3 repo
